src/esp/nav/PathFinder.h

```diff
@@ -129,60 +178,202 @@ class PathFinder : public std::enable_shared_from_this<PathFinder> {
+   *
+   * @param[in] path The saved navigation mesh file, generally has extension
+   * ``.navmesh``
+   *
+   * @return Whether or not the navmesh was successfully loaded
+   */
   bool loadNavMesh(const std::string& path);
 
+  /**
+   * @brief Saves a navigation mesh to later be loaded by @ref loadNavMesh
+   *
+   * @param[in] path The name of the file, generally has extension ``.navmesh``
+   *
+   * @return Whether or not the navmesh was successfully saved
+   */
   bool saveNavMesh(const std::string& path);
 
-  void free();
-
+  /**
+   * @return If a navigation mesh is current loaded or not
+   */
   bool isLoaded() { return navMesh_ != nullptr; }
 
+  /**
+   * @brief Seed the pathfinder.  Useful for @ref getRandomNavigablePoint
+   *
+   * @param[in] newSeed The random seed
+   *
+   * @note This just seeds the global c @ref rand function.
+   */
   void seed(uint32_t newSeed);
 
+  /**
+   * @brief returns the size of the connected component @ ref pt belongs to.
+   *
+   * @param[in] pt The point to specify the connected component
+   *
+   * @return Size of the connected component
+   */
   float islandRadius(const vec3f& pt) const;
 
+  /**
+   * @brief Finds the distance to the closest non-navigable location
+   *
+   * @param[in] pt The point to begin searching from
+   * @param[in] pt The radius to search in
+   *
+   * @return The distance to the closest non-navigable location or @ref
+   * maxSearchRadius if all locations within @ref maxSearchRadius are navigable
+   */
   float distanceToClosestObstacle(const vec3f& pt,
                                   const float maxSearchRadius = 2.0) const;
+
+  /**
+   * @brief Same as @ref distanceToClosestObstacle but returns additional
+   * information.
+   */
   HitRecord closestObstacleSurfacePoint(
       const vec3f& pt,
       const float maxSearchRadius = 2.0) const;
 
+  /**
+   * @brief Query whether or not a given location is navigable
+   *
+   * This method works by snapping @ref pt to the navigation mesh with @ref
+   * snapPoint and then checking to see if there was no displacement in the x-z
+   * plane and at most @ref maxYDelta displacement in the y direction.
+   *
+   * @param[in] pt The location to check whether or not it is navigable
+   * @param[in] maxYDelta The maximum y displacement.  This tolerance is useful
+   * for computing a top-down occupancy grid as the floor is not perfectly level
+   *
+   * @return Whether or not @ref pt is navigable
+   */
   bool isNavigable(const vec3f& pt, const float maxYDelta = 0.5) const;
 
+  /**
+   * @return The axis aligned bounding box containing the navigation mesh.
+   */
   std::pair<vec3f, vec3f> bounds() const { return bounds_; }
 
-  friend impl::ActionSpaceGraph;
-
  protected:
   bool initNavQuery();
   void removeZeroAreaPolys();
-  std::vector<vec3f> prevEnds;
+  template <typename T>
+  T tryStepImpl(const T& start, const T& end, bool allowSliding);
```

Comment:
Should this be handled via a boolean state flag? Wouldn't allowSliding always be passed as true or false for a give PathFinder instance?

Reply:
It seemed logical to just have both methods be implemented on the pathfinder and just choose the one to call.  May also be useful/let people play around with turning on/off sliding stochastically.